sw/device/tests/example_concurrency_test.c

```diff
@@ -0,0 +1,93 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * This example serves as a starting point for writing software top-level
+ * concurrency tests that use the OpenTitan Test Framework (OTTF), and run at
+ * the flash boot stage. This example is intended to be copied and modified
+ * according to the instructions below.
+ *
+ * This example demonstrates a concurrency test that spawns three FreeRTOS
+ * tasks, in addition to the `test_main` task. Each task is defined using a
+ * separate function that never returns, and deletes itself after executing its
+ * code. Since the priorities of each task are the same, yet higher than the
+ * priority of the "test_main" task, calling `ottf_task_yield()` will switch
+ * control flow between these tasks, until each task deletes itself. Then, the
+ * `test_main` task will continue executing, returning `true` when the overall
+ * test passes, triggering the OTTF to signal test execution has completed.
+ */
+
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+OTTF_DEFINE_TEST_CONFIG(.enable_concurrency = true,
+                        .can_clobber_uart = false, );
+
+static void task_1(void *task_parameters) {
+  // ***************************************************************************
+  // Place test code below.
+  // ***************************************************************************
+  LOG_INFO("Executing %s ...", ottf_task_get_self_name());
+  ottf_task_yield();
+  LOG_INFO("Continuing to execute %s ...", ottf_task_get_self_name());
+
+  // ***************************************************************************
+  // Delete the current task and never return.
+  // ***************************************************************************
+  ottf_task_delete_self();
+  abort();  // Unreachable.
```

Comment:
Nit. Could we use a `CHECK(false)` here?

Reply:
Good idea, this has been updated in the PR (#15249) this is dependent on.

The only change relevant in this PR is the last commit.